# This patch should be removed when we upgrade to 6.3.25; see the bug report
# for full context.
--- fetchmail/socket.c.orig	2012-08-13 13:02:41.000000000 -0700
+++ fetchmail/socket.c	2013-02-11 15:44:05.297674654 -0800
@@ -133,6 +133,7 @@
 	argvec = (char **)malloc(s);
 	if (!argvec)
 	{
+		free(plugin_copy);
 		report(stderr, GT_("fetchmail: malloc failed\n"));
 		return NULL;
 	}
@@ -186,6 +187,8 @@
 		if (outlevel >= O_VERBOSE)
 		    report(stderr, GT_("running %s (host %s service %s)\n"), plugin, host, service);
 		argvec = parse_plugin(plugin,host,service);
+		if (argvec == NULL)
+			_exit(EXIT_FAILURE);
 		execvp(*argvec, argvec);
 		report(stderr, GT_("execvp(%s) failed\n"), *argvec);
 		_exit(EXIT_FAILURE);
